generate-test-vector.py

```diff
@@ -1088,50 +1088,54 @@ def generate_taproot_with_nums_point_test():
         I3.negate()
 
     nums_point = [
-        [(i1, True, NUMS_H, True), (i2, True, None, False), (i3, True, NUMS_H, False)],
+        [(i1, NUMS_H, True), (i2, None, False), (i3, NUMS_H, False)],
         [I1, I2, I3]
     ]
 
     test_cases = []
     comments = [
         "Single receipient: taproot input with NUMS point"
     ]
+    included_keys = []
+    included_pubkeys = []
     for i, inputs in enumerate([nums_point]):
         sender, recipient, test_case = new_test_case()
 
         inp = []
-        for x, (key, is_taproot, internal_key, add_annex) in enumerate(inputs[0]):
+        for x, (key, internal_key, add_annex) in enumerate(inputs[0]):
             pub_key = inputs[1][x]
-            if is_taproot:
-                if (internal_key == None):
-                    inp += [{
-                        "txid": outpoints[x][0],
-                        "vout": outpoints[x][1],
-                        "scriptSig": "",
-                        "txinwitness": get_p2tr_witness(key),
-                        "prevout": {"scriptPubKey": {"hex": get_p2tr_scriptPubKey(pub_key)}},
-                    }]
-                else:
-                    inp += [{
-                        "txid": outpoints[x][0],
-                        "vout": outpoints[x][1],
-                        "scriptSig": "",
-                        "txinwitness": get_leafp2tr_witness(key, internal_key, add_annex = add_annex),
-                        "prevout": {"scriptPubKey": {"hex": get_p2tr_scriptPubKey(pub_key)}},
-                    }]
+
+            if (internal_key == None):
+                inp += [{
+                    "txid": outpoints[x][0],
+                    "vout": outpoints[x][1],
+                    "scriptSig": "",
+                    "txinwitness": get_p2tr_witness(key),
+                    "prevout": {"scriptPubKey": {"hex": get_p2tr_scriptPubKey(pub_key)}},
+                }]
+                included_keys += [(key, True)]
+                included_pubkeys += [pub_key]
             else:
+                script = "20" + pub_key.get_bytes(True).hex() + "ac"
+                leaf_version = "c0"
+                annex = "50"
+                internal_key_bytes = internal_key.to_bytes(32, 'big')
+                leaf_hash = TaggedHash("TapLeaf", bytes.fromhex(leaf_version + script))
+                tap_tweak = TaggedHash("TapTweak", internal_key_bytes + leaf_hash)
```

Comment:
You forgot to add size of script.

It should be `hash(v || compact_size(size of s) || s)`

I verified and the calculation is correct then, including the tweaked key

Reply:
thanks @S3RK ! I feel kinda silly for not spotting this earlier :sweat_smile:

Reply:
@josibake @S3RK Resolved in https://github.com/S3RK/bip-352/pull/5/commits/4e5b3c637b712961b0ee8cd8c876ef661b8b70c8